packages/xchain-radix/src/client.ts

```diff
@@ -0,0 +1,482 @@
+  TRANSACTION_PREVIEW_PATH,
+  TRANSACTION_SUBMIT_PATH,
+  TransferTransactionManifest,
+  XRD_DECIMAL,
+} from './const'
+import { EntityDetailsResponse, Transaction } from './types/radix'
+
+/**
+ * Custom Radix client
+ */
+
+class Client extends BaseXChainClient {
+  privateKey: PrivateKey
+  gatewayApiClient: GatewayApiClient
+  constructor(params: XChainClientParams, keyType: KeyType) {
+    super(RadixChain, { network: params.network, phrase: params.phrase })
+    const seed = getSeed(this.phrase)
+    const hexString = seed.toString('hex').substring(0, 64)
+    if (keyType == KeyType.Ed25519) {
+      this.privateKey = new PrivateKey.Ed25519(hexString)
+    } else {
+      this.privateKey = new PrivateKey.Secp256k1(hexString)
+    }
+    this.gatewayApiClient = GatewayApiClient.initialize({
+      networkId: this.getRadixNetwork(),
+      applicationName: 'xchainjs',
+    })
+  }
+  /**
+   * Get an estimated fee for a test transaction that involves sending
+   * XRD from one account to another
+   *
+   * @returns {Fee} An estimated fee
+   */
+  async getFees(): Promise<Fees> {
+    const transactionPreviewUrl = `${this.getGatewayUrl()}${TRANSACTION_PREVIEW_PATH}`
+    const transactionConstructionUrl = `${this.getGatewayUrl()}${TRANSACTION_CONSTRUCTION_PATH}`
+    try {
+      const constructionMetadataResponse = await axios.post(transactionConstructionUrl)
+      const transaction: Transaction = {
+        manifest: TransferTransactionManifest,
+        start_epoch_inclusive: constructionMetadataResponse.data.ledger_state.epoch,
+        end_epoch_exclusive: constructionMetadataResponse.data.ledger_state.epoch + 10,
+        tip_percentage: 10,
+        nonce: Math.floor(Math.random() * 1000000),
+        signer_public_keys: [
+          {
+            key_type: 'EddsaEd25519',
+            key_hex: this.privateKey.publicKey().hex(),
+          },
+        ],
+        flags: {
+          use_free_credit: true,
+          assume_all_signature_proofs: false,
+          skip_epoch_check: true,
+        },
+      }
+      const response = await axios.post(transactionPreviewUrl, transaction)
+      const previewTransaction = response.data.receipt
+      let totalFees = 0
+      for (const key in previewTransaction.fee_summary) {
+        if (!isNaN(parseFloat(previewTransaction.fee_summary[key])) && key.startsWith('xrd_total_')) {
+          totalFees += parseFloat(previewTransaction.fee_summary[key])
+        }
+      }
+      const estimatedFees: Fees = {
+        type: FeeType.FlatFee,
+        [FeeOption.Average]: assetToBase(assetAmount(totalFees)),
+        [FeeOption.Fast]: assetToBase(assetAmount(totalFees)),
+        [FeeOption.Fastest]: assetToBase(assetAmount(totalFees)),
+      }
+      return estimatedFees
+    } catch (error) {
+      throw new Error('Failed to calculate the fees')
+    }
+  }
+
+  getRadixNetwork(): number {
+    const network = this.getNetwork()
+    let networkId: number
+    switch (network) {
```

Comment:
Looking at the other clients this does not seem to be true: [example1](https://github.com/radixdlt/xchainjs-lib/blob/e0d9669c549e0899fc30b4251018cebdac6768ca/packages/xchain-arbitrum/src/const.ts#L30-L34), [example2](https://github.com/radixdlt/xchainjs-lib/blob/e0d9669c549e0899fc30b4251018cebdac6768ca/packages/xchain-avax/src/const.ts#L21-L25).

So, I think that the correct thing here would be:

* `Network.Mainnet | Network.Stagenet` => `NetworkId.Mainnet`.
* `Network.TestNet` => `NetworkId.Stokenet`.

I do not think that an internal test net is needed at all here. In fact, I'm not certain what the gateway client will do if encountered with such a network id and how it would find the base url for the gateway api.

Reply:
@balda-rdx Ping on the above